### ISSUE TITLE
Fake Grasping Problem

### ISSUE THREAD

**Issue by robfiras** [stepjam/RLBench#105]
As can be seen, there is a problem with fake grasping. The agents learns a bad policy (it seems to be easier to learn fake grasping).
![Problem_Fake_Grasping2](https://user-images.githubusercontent.com/69359729/105926609-a5c21480-6042-11eb-8354-138ffc8aca9e.gif)

I have read this #97, but setting `attach_grasped_objects=False` seems to turn off the rewards also. The latter wouldn't make sense in my eyes. Is there a possibility to set `attach_grasped_objects=False` but keep the reward?

**Comment by stepjam** [comment]
Yes, fake grasping will indeed make the task easier, as it makes it easy for the policy to exploit this.
Indeed, setting `attach_grasped_objects=False` should not disable the reward; I'll take a look and report back.

**Comment by stepjam** [comment]
After taking a quick look, It's not obvious to me what the reward would be disabled...
What exactly do you mean by the reward being disabled? How are you checking this?
have you shaped the reward to learn this policy? Or is it learned from the given sparse reward?

**Comment by robfiras** [comment]
Thanks for the quick reply!
Sorry for my inaccurate description, it was late already.. 
So what I meant was that this particular task (derivate of "pick and lift") uses the `GraspedCondition` as a part if its reward. The latter checks for grasped objects at the gripper. However, the gripper will never grasp an object if `attach_grasped_objects=False`. So the `GraspedCondition`  won't be me. Or am I wrong?

This policy is learned by shaped rewards, including rewards given by (or based on) the `GraspedCondition`.
